transform/transformers/PDFTransformer.py

```diff
@@ -77,33 +96,68 @@ def get_elements(self):
 
-        localised_date_str = self.get_localised_date(self.response['submitted_at'])
+        localised_date_str = PDFTransformer.get_localised_date(response['submitted_at'])
 
-        heading_data = [[Paragraph(self.survey['title'], styleH)]]
-        heading_data.append(['Form Type', self.response['collection']['instrument_id']])
-        heading_data.append(['Respondent', self.response['metadata']['ru_ref']])
+        heading_data = [[Paragraph(survey['title'], styleH)]]
+        heading_data.append(['Form Type', response['collection']['instrument_id']])
+        heading_data.append(['Respondent', response['metadata']['ru_ref']])
         heading_data.append(['Submitted At', localised_date_str])
 
         heading = Table(heading_data, style=heading_style, colWidths='*')
 
         elements.append(heading)
 
-        for question_group in self.survey['question_groups']:
+        for question_group in survey['question_groups']:
 
             if 'title' in question_group:
-                # meta = question_group['meta'] if 'meta' in question_group else ''
                 elements.append(Paragraph(question_group['title'], styleSH))
 
                 for question in question_group['questions']:
                     if 'text' in question:
                         answer = ''
-                        if question['question_id'] in self.response['data']:
-                            answer = self.response['data'][question['question_id']]
+                        if question['question_id'] in response['data']:
+                            answer = response['data'][question['question_id']]
 
-                        elements.append(Paragraph(question['text'], styleSSH))
+                        text = question.get("text", " ")
+                        if not text[0].isdigit():
+                            text = " ".join((question.get("number", ""), text))
+                        elements.append(Paragraph(text, styleSSH))
                         elements.append(Paragraph(answer, styleN))
 
         return elements
 
-    def get_localised_date(self, date_to_transform, timezone='Europe/London'):
+    @staticmethod
+    def get_localised_date(date_to_transform, timezone='Europe/London'):
         return arrow.get(date_to_transform).to(timezone).format("DD MMMM YYYY HH:mm:ss")
+
+
+def parser(description=__doc__):
+    rv = argparse.ArgumentParser(
+        description,
+    )
+    rv.add_argument(
+        "--survey", required=True,
+        help="Set a path to the survey JSON file.")
+    return rv
+
+
+def main(args):
+    fP = os.path.expanduser(os.path.abspath(args.survey))
+    with open(fP, "r") as fObj:
+        survey = json.load(fObj)
+
+    data = json.load(sys.stdin)
+    tx = PDFTransformer(survey, data)
+    output = tx.render()
+    sys.stdout.write(output.decode("latin-1"))
+    return 0
+
+
+def run():
+    p = parser()
+    args = p.parse_args()
+    rv = main(args)
+    sys.exit(rv)
```

Comment:
I do it this way since reading this:
http://www.artima.com/weblogs/viewpost.jsp?thread=4829

Obviously, we're using argparse here, etc.